src/hooks/useGrantContract.ts

```diff
@@ -0,0 +1,70 @@
+import { ethers } from 'ethers';
+import { useMemo, useState } from 'react';
+import { TokenVestingController__factory } from 'contracts/types';
+
+import { useWeb3React } from '@web3-react/core';
+
+import { GrantContract, Maybe } from '../util';
+
+enum errorState {
+	NoError = 'Checking...',
+	NotOwner = 'You are not the Owner of this address.',
+	NotChainId = 'No chain id detected.',
+	NotSuportChainId = 'Chain id not supported.',
+	NoAddress = 'Invalid vesting contract address.',
+}
+
+export function useGrantContract(address: string) {
+	const context = useWeb3React<ethers.providers.Web3Provider>();
+
+	const [isError, setError] = useState('');
```

Comment:
`isError` implies that `Error` is a boolean value; "is there an error", consider renaming this to just `error` since the setter is `setError`.

As a rule of thumb, whenever you create a React State Variable (using `React.useState`) it should always follow this format: `[X, setX]` where `X` is the name of the variable.